src/n_body_simulation.py

```
@@ -83,16 +86,20 @@ def add_solar_system(self, date=None):
-                         # Fallback label
-                         p.label = f"particle_{p.index}"
+
+            # Assign labels for easier access using index-based mapping
+            # REBOUND's add_solar_system adds particles with names like 'Sun', 'Jupiter', etc.
+            for i, p in enumerate(self.sim.particles):
+                 # Try to get the name assigned by REBOUND
+                 try:
+                     name = p.name  # REBOUND assigns names to solar system bodies
+                     if name:
+                         self._label_to_index[name] = i
+                 except:
```

Ruff (0.15.20)

[error] Do not use bare `except`

(E722)